### PR TITLE
Use stable selection of primary vs verify

### PR DESCRIPTION
**Pull request by mdegerne**
By basing the selection on the name rather than the ROWID, we avoid items
being missed because of differences in database ordering.

### CODE REVIEW

**Comment by mdegerne** [comment]
As an FYI, i ran this algorithm over /usr/share/dict and here is the histogram of % 10:

```>>> with open('/usr/share/dict/words') as f:
...   for w in f.readlines():
...     l[hnum(w.strip()) % 10] += 1
... 
>>> l
[23738, 23684, 23832, 23424, 23630, 23383, 23622, 23592, 23623, 23360]
```

File: container_crawler/crawler.py [hunk 1]
```diff
@@ -70,6 +71,10 @@ def complete_task(self, error=False, retry=False):
                 self._done.notify()
 
 
+def num_from_row(row):
+    return int(hashlib.sha1(row['name']).hexdigest()[-6:], 16)
```
Comment (timuralp) [inline container_crawler/crawler.py]: It probably doesn't matter how many digits we grab (or if we even use the whole hash output), but let's turn the 6 into a constant somewhere?

**Comment by timuralp** [comment]
@mdegerne also, cool! Thanks for confirming it's uniform :)

File: container_crawler/crawler.py (hunk 1 above)
Comment (mdegerne) [reply container_crawler/crawler.py]: Added constant.

**Comment by timuralp** [comment]
I don't understand how the new function to hash the object names doesn't get covered, but I don't see that as a blocker.

**Pull request merged**